Combined and sex-stratified genome-wide association meta-studies performed in a bash environment using  MetaGWASToolKit. Post-GWAS analysis is performed using clumping with PLINK and gene annotation with FUMA. 

Scripts used for meta-analysis can be found in the [MetaGWASToolKit](https://github.com/swvanderlaan/MetaGWASToolKit.git) repository. 
Scripts used for clumping, the miami plot, blobbogram and interaction-analysis are not from MetaGWASToolKit and are found in this repository.

<br><br><br>
![Alt text](https://github.com/xEmz/UMC-GWAS-cIMT/blob/97f50fb5532a0758e47736762cc6b63b54be4ccc/Meta-analysis/cIMT-meta.drawio.png)

</br></br></br>
# Table of contents


- [SCRIPTS](https://github.com/xEmz/UMC-GWAS-cIMT/blob/2f335d6718aee6cea9715422783da142525b7c0d/Meta-analysis/cIMT-meta.drawio.png)

- [FLOWCHART](https://github.com/xEmz/UMC-GWAS-cIMT/blob/97f50fb5532a0758e47736762cc6b63b54be4ccc/Meta-analysis/cIMT-meta.drawio.png)

- [Information on the cohorts used in the meta-analysis](https://github.com/xEmz/UMC-GWAS-cIMT/blob/main/Meta-analysis/Cohort_Data.xlsx)
