Combined and sex-stratified genome-wide association meta-studies performed in a bash environment using  MetaGWASToolKit. Post-GWAS analysis is performed using clumping with PLINK and gene annotation with FUMA. 

Scripts used for meta-analysis can be found in the [MetaGWASToolKit](https://github.com/swvanderlaan/MetaGWASToolKit.git) repository.

<br>

![Alt text](https://github.com/xEmz/Avans-files/blob/9d52a2af6d63e27dfbadc65b587d69f7eccc266c/cIMT-meta.drawio.png)

</br>

# Table of contents

The data of the cohorts, including the study name and database, sample size of females and males, case status, ancestry, genotyping method, reference genome, imputation method and reference dataset, GWAS method and	cIMT measurement.

- [SCRIPTS](https://github.com/xEmz/UMC-GWAS-cIMT/blob/9282e2a2d96468699205437dc824ce91e6195157/SMART-GWAS/SCRIPTS)
