Combined and sex-stratified genome-wide association meta-studies performed in a bash environment using  MetaGWASToolKit. Post-GWAS analysis is performed using clumping with PLINK and gene annotation with FUMA. 

Scripts used for meta-analysis can be found in the [MetaGWASToolKit](https://github.com/swvanderlaan/MetaGWASToolKit.git) repository.

<br>

![Alt text](https://github.com/xEmz/Avans-files/blob/9d52a2af6d63e27dfbadc65b587d69f7eccc266c/cIMT-meta.drawio.png)

</br>
