Combined and sex-stratified genome-wide association meta-studies performed in a bash environment using  MetaGWASToolKit. Post-GWAS analysis is performed using clumping with PLINK and gene annotation with FUMA. 

Scripts used for meta-analysis can be found in the [MetaGWASToolKit](https://github.com/swvanderlaan/MetaGWASToolKit.git) repository. 
Scripts used for the miami plot and interaction-analysis are not included in MetaGWASToolKit and are found in this repository.

<br><br><br>
![Alt text](https://github.com/xEmz/Avans-files/blob/0b5db23474547033b8c6499c166b537ac2e6075b/cIMT-meta.drawio.png)

</br></br></br>
# Table of contents


- [SCRIPTS](https://github.com/xEmz/UMC-GWAS-cIMT/tree/main/Meta-analysis/SCRIPTS)
-[Information on the cohorts used in the meta-analysis](https://github.com/xEmz/UMC-GWAS-cIMT/blob/main/Meta-analysis/Cohort_Data.xlsx)
